holoviews/plotting/mpl/element.py

```diff
@@ -694,7 +694,13 @@ def _norm_kwargs(self, element, ranges, opts, vdim):
             self._cbar_extend = 'max'
 
         # Define special out-of-range colors on colormap
-        cmap = copy.copy(plt.cm.get_cmap(opts.get('cmap')))
+        cmap = opts.get('cmap')
+        if isinstance(cmap, list):
+            cmap = mpl_colors.ListedColormap(cmap)
+        elif isinstance(cmap, util.basestring):
+            cmap = copy.copy(plt.cm.get_cmap(cmap))
+        else:
+            cmap = copy.copy(cmap)
```

Comment:
What is all the copying about?

Reply:
I modify the colormaps to set the NaN, over and under colors and we definitely don't want to modify the default colormaps or user supplied ones.